packages/backend/src/models/NotificationsModel/NotificationsModel.ts

```diff
@@ -0,0 +1,125 @@
+            })[] = await this.database(NotificationsTableName)
+            .join(
+                UserTableName,
+                `${NotificationsTableName}.user_uuid`,
+                '=',
+                `${UserTableName}.user_uuid`,
+            )
+            .join(
+                DashboardTileCommentsTableName,
+                `${NotificationsTableName}.comment_id`,
+                '=',
+                `${DashboardTileCommentsTableName}.comment_id`,
+            )
+            .join(
+                `${UserTableName} as comment_authors`,
+                `${DashboardTileCommentsTableName}.user_uuid`,
+                '=',
+                'comment_authors.user_uuid',
+            )
+            .join(
+                DashboardsTableName,
+                `${NotificationsTableName}.dashboard_uuid`,
+                '=',
+                `${DashboardsTableName}.dashboard_uuid`,
+            )
+            .select(
+                `${NotificationsTableName}.*`,
+                `${UserTableName}.first_name as user_first_name`,
+                `${UserTableName}.last_name as user_last_name`,
+                `${DashboardsTableName}.name as dashboard_name`,
+                `${DashboardTileCommentsTableName}.dashboard_tile_uuid`,
+                'comment_authors.first_name as author_first_name',
+                'comment_authors.last_name as author_last_name',
+            )
+            .where(`${NotificationsTableName}.user_uuid`, userUuid)
+            .orderBy(`${NotificationsTableName}.created_at`, 'desc');
+
+        return notificationsWithUserAndDashboardAndAuthor.map((notif) => ({
+            notificationId: notif.notification_id,
+            user: {
+                name: `${notif.user_first_name} ${notif.user_last_name}`,
+            },
+            author: {
+                name: `${notif.author_first_name} ${notif.author_last_name}`,
+            },
+            dashboard: {
+                uuid: notif.dashboard_uuid,
+                name: notif.dashboard_name,
+                tileUuid: notif.dashboard_tile_uuid,
+            },
+            viewed: notif.viewed,
+            createdAt: notif.created_at,
+        }));
+    }
+
+    async markNotificationAsRead(notificationUuid: string) {
+        return this.database('notifications')
+            .where({ notification_id: notificationUuid })
+            .update({ viewed: true });
+    }
+
+    async createDashboardCommentNotification(
+        userUuid: string,
+        commentId: string,
+        dashboardUuid: string,
+    ) {
+        const comment = await this.database(DashboardTileCommentsTableName)
+            .where({ comment_id: commentId })
+            .first();
+
+        if (!comment) {
+            throw new Error('Comment not found');
+        }
+
+        if (comment.mentions.length > 0) {
+            await Promise.all(
+                comment.mentions.map((mentionUserUuid) =>
+                    this.database.transaction(async (trx) => {
```

Comment:
I don't think this transaction is useful here. because it only has one operation inside . So you can either replace the transaction with  a database.insert or move the transaction outside of the map

Reply:
absolutely! I had 2 more trxs (from the old base PR) but we don't need them. 
removed this transaction